community/lucene-index/src/main/java/org/neo4j/kernel/api/impl/schema/writer/PartitionedIndexWriter.java

```diff
@@ -34,16 +34,18 @@
  * Schema Lucene index writer implementation that supports writing into multiple partitions and creates partitions
  * on-demand if needed.
  * <p>
- * Writer threats partition as writable if partition has number of documents that is less then configured
+ * Writer threats partition as writable if partition has number of live and deleted documents that is less then configured
  * {@link #MAXIMUM_PARTITION_SIZE}.
  * First observable partition that satisfy writer criteria is used for writing.
  */
 public class PartitionedIndexWriter implements LuceneIndexWriter
 {
     private final WritableDatabaseSchemaIndex index;
 
+    // by default we still keep a spare of 10% to the maximum partition size: During concurrent updates
```

Comment:
We keep this big (comparatively) gap to avoid synchronisation in here. It's fine if we will expose a writer even in quota is reached or almost reached. Since population will gonna happen in batches and if couple of them will gonna squeeze is, it's fine. Thats why we can avoid any atomics or complete sync.